akka-actor-typed/src/main/scala/akka/actor/typed/ActorLogger.scala

```diff
@@ -0,0 +1,548 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.actor.typed
+
+import akka.annotation.{ DoNotInherit, InternalApi }
+
+/**
+ * A log marker is an additional metadata tag supported by some logging backends to identify "special" log events.
+ * In the Akka internal actors for example the "SECURITY" marker is used for warnings related to security.
+ *
+ * Not for user extension
+ */
+@DoNotInherit
+sealed trait LogMarker {
+  def name: String
+}
+
+/**
+ * Factories for log markers
+ */
+object LogMarker {
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  private final class LogMarkerImpl(name: String) extends akka.event.LogMarker(name) with LogMarker
+
+  /**
+   * Scala API: Create a new log marker with the given name
+   */
+  def apply(name: String): LogMarker = new LogMarkerImpl(name)
+
+  /**
+   * Scala API: Create a new log marker with the given name
+   */
+  def create(name: String): LogMarker = apply(name)
+
+}
+
+/**
+ * Logging API provided inside of actors through the actor context.
+ *
+ * All log-level methods support simple interpolation templates with up to four
+ * arguments placed by using <code>{}</code> within the template (first string
+ * argument):
+ *
+ * {{{
+ * ctx.log.error(exception, "Exception while processing {} in state {}", msg, state)
+ * }}}
+ *
+ * More than four arguments can be defined by using an `Array` with the method with
+ * one argument parameter.
+ *
+ * *Rationale for an Akka-specific logging API:*
+ * Provided rather than a specific logging library logging API to not enforce a specific logging library on users but
+ * still providing a convenient, performant, asynchronous and testable logging solution. Additionally it allows unified
+ * logging for both user implemented actors and built in Akka actors where the actual logging backend can be selected
+ * by the user.
+ *
+ * The [[ActorLogger]] of an actor is tied to the actor path and should not be shared with other threads outside of
+ * the actor.
+ *
+ * Not for user extension
+ */
+@DoNotInherit
+abstract class ActorLogger private[akka] () {
```

Comment:
I had a previous comment about the name ActorLogger?

Reply:
I think that got lost somehow, I saw a notification but when I clicked the link I couldn't see any comment from you, please repeat it!

Reply:
If we (later) decide that it should be possible to use this outside actors also then the name is not good. Similar to how a `LoggingAdapter` can be created for a class (given an ActorSystem).

I'd like `Logger` but that is already taken, I think.

Reply:
Can we hijack "Logging"?

Reply:
I think `Logger` is good, that's what everyone other log interface is called (`j.u.l.Logger`, `org.apache.logging.log4j.Logger`, `org.slf4j.Logger`, etc), and we have the package namespace - plus the type won't really be referenced much so risk for conflict is small.

Reply:
Yes good point... and people usually know WHICH one they wanna use anyway. Sounds good to me.